src/prof_recent.c

```diff
@@ -556,5 +662,18 @@ prof_recent_init() {
 	}
 	prof_recent_alloc_dummy->next = prof_recent_alloc_dummy;
 
+	assert(prof_recent_alloc_dummy_reserve == NULL);
+	prof_recent_alloc_dummy_reserve = (prof_recent_t *)iallocztm(
```

Comment:
Abstract to a function allocating the node? If the only difference between this and the regular node allocation is the TSDN_NULL, it can be an input arg.

Reply:
One more subtle difference: I'm here using `arena_get(TSDN_NULL, 0, true)`, whereas for prof allocations I'm using `arena_get(tsd_tsdn(tsd), 0, false)`. In other words I force create the arena in boot but not in real traffic. Do we want to change anything here?

Reply:
I think the `true` for force creating isn't really necessary for the boot case? prof_recent should be initialized late enough.

Reply:
Makes sense. Abstracted out the allocation logic. Also abstracted out the deallocation logic to be symmetric.